tests/test_venv.py

```diff
@@ -265,6 +265,31 @@ def test_develop_extras(newmocksession, tmpdir):
     expected = "%s[testing,development]" % tmpdir.strpath
     assert expected in l[-1].args
 
+def test_env_variables_added_to_needs_reinstall(tmpdir, mocksession, newconfig, monkeypatch):
+    tmpdir.ensure("setup.py")
+    monkeypatch.setenv("TEMP_PASS_VAR", "123")
+    monkeypatch.setenv("TEMP_NOPASS_VAR", "456")
+    config = newconfig([], """
+        [testenv:python]
+        passenv = temp_pass_var
+        setenv =
+            CUSTOM_VAR = 789
+    """)
+
+    venv = VirtualEnv(config.envconfigs['python'], session=mocksession)
+    action = mocksession.newaction(venv, "hello")
+
+    venv._needs_reinstall(tmpdir, action)
+
+    l = mocksession._pcalls
+    assert len(l) == 1
+    env = l[0].env
+    assert 'CUSTOM_VAR' in env
+    assert env['CUSTOM_VAR'] == '789'
+    assert 'TEMP_PASS_VAR' in env
+    assert env['TEMP_PASS_VAR'] == "123"
+    assert 'TEMP_NOPASS_VAR' in env
```

Comment:
I'm not familiar with tox's overall structure, but that seemed more consistent with the old behavior (since it was previously being invoked under the outside environment).  And it's inline with the venv.run_install_command() -- which invokes venv._pcall() in such a way that it's *also* called with the full outside environment + setenv vars.   Per venv._getenv(), it looked to me like only the test commands themselves get the restricted environment.